kubernetes_platform/python/kfp/kubernetes/secret.py

```diff
@@ -75,11 +76,16 @@ def use_secret_as_volume(
     Returns:
         Task object with updated secret configuration.
     """
-
     msg = common.get_existing_kubernetes_config_as_message(task)
 
+    val = secret_name
+    # if secret_name is a PipelineParameterChannel, then we don't know what secret to mount until RUNTIME
+    # so, treat is as a map KEY instead of a secret name
+    if isinstance(secret_name, PipelineParameterChannel):
+        val = "{{" + secret_name.name + "}}"
```

Comment:
I am uncertain about this approach, we're essentially implementing templating, but what seems like a hacky way. Have we considered other approaches? 

For example, we for names retrieved via parameterschannels, we can include a new field for `SecretAsEnv` called "secret_name_parameter". 

Based on last community call @chensun had a proposal for an alternative approach as well, so I would be interested in hearing about this as well.

Reply:
> we can include a new field for SecretAsEnv called "secret_name_parameter".

that _does_ feel cleaner to me